scikeras/wrappers.py

```diff
@@ -51,102 +41,67 @@ class BaseWrapper(BaseEstimator):
-    3. An instance of a Keras Model. A copy of this instance will be made.
-    4. None (the default value). This means you implement a class that inherits
-    from `BaseWrapper`, `KerasClassifier` or `KerasRegressor`.
-    The `_keras_build_fn` method of this class must return an instance of a
-    Keras Model.
-
-    The parameters passed to `build_fn` or `_keras_build_fn` are dynamically
-    determined based on the signature of `build_fn` or `_keras_build_fn`.
-    All public attributes as well as any `**kwargs` passed to `fit` will be
-    passed to  `build_fn` or `_keras_build_fn` if their name matches one of
-    the parameter names in `build_fn` or `_keras_build_fn`.
-
-    `sk_params` takes both model parameters and fitting parameters.
-    Note that like all other estimators in scikit-learn, `build_fn` or
-    your child class should provide default values for its arguments,
-    so that you could create the estimator
-    without passing any values to `sk_params`.
-
-    `sk_params` could also accept parameters for calling `fit`, `predict`,
-    `predict_proba`, and `score` methods (e.g., `epochs`, `batch_size`).
-    fitting (predicting) parameters are selected in the following order:
-
-    1. Values passed to the dictionary arguments of
-    `fit`, `predict`, `predict_proba`, and `score` methods
-    2. Values passed to `sk_params`
-    3. The default values of the `keras.models.Sequential`
-    `fit`, `predict`, `predict_proba` and `score` methods
-
-    When using scikit-learn's `grid_search` API, legal tunable parameters are
-    those you could pass to `sk_params`, including fitting parameters.
-    In other words, you could use `grid_search` to search for the best
-    `batch_size` or `epochs` as well as the model parameters.
+            Used to build the Keras Model. When called,
+            must return a compiled instance of a Keras Model
+            to be used by `fit`, `predict`, etc.
+        random_state : int, RandomState instance, default=None
+            Set the Tensorflow random number generators to a
+            reproducible deterministic state using this seed.
+            Pass an int for reproducible results across multiple function calls.
+        For all other parameters see tf.keras.Model documentation.
     """
 
-    # basic legal parameter set, based on functions that will normally be
-    # called the model building function will be dynamically added
-    _legal_params_fns = [
-        Sequential.evaluate,
-        Sequential.fit,
-        Sequential.predict,
-        Model.evaluate,
-        Model.fit,
-        Model.predict,
-    ]
-
-    _sk_params = None
     is_fitted_ = False
 
     _tags = {
         "poor_score": True,
         "multioutput": True,
     }
 
-    def __init__(self, build_fn=None, **sk_params):
+    def __init__(
+        self,
+        build_fn=None,
+        *,
+        random_state=None,
+        optimizer="rmsprop",
+        loss=None,
```

Comment:
Where are `optimizer` and `loss` used? Are they passed to the `compile` method?

Reply:
As it stands right now, it would be up to the user to declare their `build_fn` to use them. I.e.:

```python3
def build_fn(optimizer=..., loss=...):
    model = Model()
    ...
    model.compile(loss=loss, optimizer=optimizer)
```